node/src/node/isometrics.hpp

```diff
@@ -0,0 +1,278 @@
+        asio::io_service& loop,
+        const std::uint64_t poll_interval);
+
+    ///
+    /// Reads following section from Cocaine-RT configuration:
+    ///  ```
+    ///  "node" : {
+    ///     ...
+    ///     "args" : {
+    ///        "isolate_metrics:" : true,
+    ///        "isolate_metrics_poll_period_s" : 10
+    ///     }
+    ///  }
+    ///  ```
+    /// if `isolate_metrics` is false (default), returns nullptr,
+    /// and polling sequence wouldn't start.
+    ///
+    // TODO: it seems that throw on construction error would be a better choice.
+    template<typename Observers>
+    static
+    auto
+    make_and_ignite(
+        context_t& ctx,
+        const std::string& name,
+        std::shared_ptr<api::isolate_t> isolate,
+        synchronized<engine_t::pool_type>& pool,
+        asio::io_service& loop,
+        synchronized<Observers>& observers) -> std::shared_ptr<metrics_retriever_t>;
+
+    auto
+    ignite_poll() -> void;
+
+    auto
+    make_observer() -> std::shared_ptr<pool_observer>;
+
+    //
+    // Should be called on every pool::erase(id) invocation
+    //
+    // usually isolation daemon will hold metrics of despawned workers for some
+    // reasonable period (at least 30 sec), so it is allowed to request metrics
+    // of dead workers on next `poll` invocation.
+    //
+    // Note: on high despawn rates stat of some unlucky workers will be lost
+    //
+    auto
+    add_post_mortem(const std::string& id) -> void;
+
+private:
+
+    auto
+    poll_metrics(const std::error_code& ec) -> void;
+
+private:
+
+    // TODO: wip, possibility of redesign
+    struct metrics_handle_t : public api::metrics_handle_base_t
+    {
+        metrics_handle_t(std::shared_ptr<metrics_retriever_t> parent) :
+            parent{parent}
+        {}
+
+        auto
+        on_data(const dynamic_t& data) -> void override;
+
+        auto
+        on_error(const std::error_code&, const std::string& what) -> void override;
+
+        std::shared_ptr<metrics_retriever_t> parent;
+    };
+
+    // TODO: wip, possibility of redesign
+    struct metrics_pool_observer_t : public pool_observer {
+
+        metrics_pool_observer_t(metrics_retriever_t& p) :
+            parent(p)
+        {}
+
+        auto
+        spawned(const std::string&) -> void override
+        {}
+
+        auto
+        despawned(const std::string& id) -> void override {
+            parent.add_post_mortem(id);
+        }
+
+    private:
+        metrics_retriever_t& parent;
+    };
+
+}; // metrics_retriever_t
+
+template<typename Observers>
+auto
+metrics_retriever_t::make_and_ignite(
+    context_t& ctx,
+    const std::string& name,
+    std::shared_ptr<api::isolate_t> isolate,
+    synchronized<engine_t::pool_type>& pool,
+    asio::io_service& loop,
+    synchronized<Observers>& observers) -> std::shared_ptr<metrics_retriever_t>
+{
+    const auto node_config = ctx.config().component_group("services").get("node");
+
+    if (!node_config) {
```

Comment:
is this a good idea to return nullptr? Besides for some(very strange) reasons node service can have a different name (not 'node').